### DISCUSSION TITLE
Deploying Tactical RMM over GPO

### DISCUSSION THREAD

**Discussion by i7phillip** [amidaware/tacticalrmm#1403] [Q&A]
I know this has been asked before but that thread is old and did not answer any of my questions. 

I am trying to deploy tactical over GPO but I am not having any luck. (I did see that over a year ago, it was said that .msi was coming to TRMM but I haven't seen it yet).

Here's my powershell script file:

```
Invoke-WebRequest "my_url" -OutFile ( New-Item -Path "c:\TacticalRMM\temp\trmminstall.exe" -Force )
$proc = Start-Process c:\TacticalRMM\temp\trmminstall.exe -ArgumentList '-silent' -PassThru
Wait-Process -InputObject $proc

if ($proc.ExitCode -ne 0) {
    Write-Warning "$_ exited with status code $($proc.ExitCode)"
}
}
Remove-Item -Path "c:\TacticalRMM\temp\trmminstall.exe" -Force
```
I would really like to deploy this when a user logs in to the domain. I have applied this script to GPO > User Config > Windows Settings > Scripts > Logon

I am thinking that I need this script to wait for network before running, but I'm not sure how to add that to my script.

Thanks.

**Comment by dinger1986** [comment]
Are your users admins?

Have you made AV exclusions?

**Reply by i7phillip** [reply]
Users are not admins.

I have made an AV exclusion to the c:\path

**Reply by dinger1986** [reply]
If they aren't admins it'll not install. You could use something like http://easydeploy.andyamaya.com. Usually I recommend fepstools but the site appears to be down for that.

Did you make sure it's allowing all the necessary exclusions for AV?

**Reply by i7phillip** [reply]
Sorry, users are admins. Even if they weren't, I should be able to use GPO on Computer Policy to force the install. Afterall, Tactical shows it can be done, but doesn't explain it very well.

**Reply by dinger1986** [reply]
Not as a startup script as thats using the user session not the admin session, but if the users are admins then this it moot. Presumably you mean this https://docs.tacticalrmm.com/install_agent/#scripting-agent-installation? Intended to use this method https://learn.microsoft.com/en-us/troubleshoot/windows-server/group-policy/use-group-policy-to-install-software. 

Can you install the agents directly?

Can you install the agents using a tools like pstools or something on machines?

If the answer to either of these is no then its probably AV exclusions needing set, if the answer is yes then its something with your deployment.

**Reply by silversword411** [reply]
You've reviewed https://docs.tacticalrmm.com/install_agent/#scripting-agent-installation

The MSI, and software deployment?
